src/roiextractors/extractors/hdf5imagingextractor/hdf5imagingextractor.py

```diff
@@ -100,7 +100,9 @@ def get_frames(self, frame_idxs: ArrayType, channel: Optional[int] = 0):
             slice_start = 0
             slice_stop = self.get_num_frames()
 
-        frames = self._video.lazy_slice[slice_start:slice_stop, :, :, channel]
+        frames = self._video.lazy_slice[slice_start:slice_stop, :, :, channel].dsetread()
+        if isinstance(frame_idxs, int):
+            frames = frames.squeeze()
```

Comment:
Just to confirm, I had thought that the goal here was to keep the inner frame dimension of `(1,num_rows,num_columns)` in the event `frame_idxs=scalar`? Won't the squeeze reduce it to `(num_rows, num_cols)`?

Reply:
Numpy does not return the axis corresponding to the scalar when indexed by it:
https://github.com/catalystneuro/roiextractors/issues/147

Maybe there was a misunderstanding?

Reply:
Looks like a misunderstainding indeed, I did not check what numpy naturally does in that case, I only checked what SpikeInterface does (which keeps the axis in-place). I was saying we should mimic SpikeInterface as much as possible, but I didn't know SpikeInterface wasn't aligned with numpy

Reply:
Interestingly the decorator  `check_get_frames_args` eliminates the axis in both cases.

Reply:
I am coming back to this. I am now confused by how you were thinking concerning spikeinterface. As `spikeinterface` uses slice semantics, lists with a single element do not make sense over there.

Reply:
@h-mayorquin The return structure when the slice spawns only a single element keeps the axis in place both for frames and channels (no squeezing)

Reply:
I think @weiglszonja would also benefit from that behavior with her new ImagingDataChunkIterator, as it was an unnecessary complexity to have to add force a new-axis when/if iterating over one axis at a time (important for the iterator's returned data to have the same structure as the selection)

Reply:
Isn't the new `ImagingDataChunkIterator` working with `get_video` instead of with `get_frames`? Get video has the desired behavior so this should not be a problem  anymore.

Reply:
@weiglszonja Can you confirm this?

The relevant lines from her PR: https://github.com/catalystneuro/nwb-conversion-tools/pull/575/files#diff-3419e31114e44476c4474658e8b2892e041cdcf50db4c99176d0d17a18e4fb67R110-R117

but have not tested it myself

Reply:
But yes, we are transitioning to `get_video` thankfully, so that is the place to fix it now (if still as issue)

Reply:
Yes, if you want that the iterator behaves as similar to the one that you use for spikeinterface then you should use `get_video`. 

The tests for the desired behavior should be working here. Please take a look.
https://github.com/catalystneuro/roiextractors/pull/181